### PR TITLE
Rebuild

### PR DESCRIPTION
**Pull request by tardyp**
fixes: http://trac.buildbot.net/ticket/2639

as there is no add Buildrequest api, I tried to reuse as much as possible the addBuildSet api, thus we don't share the buildset, nor properties, but we share the sourcestamp

### CODE REVIEW

**Comment by Stibbons** [comment]
What about rebuild with property edition (some talked this noon about this "killing" feature)?

**Comment by tardyp** [comment]
Yes this is only valid for force build which are not so common.

It's actually a much easier story. Just ui..

Le lun. 7 sept. 2015 20:05, Gaetan Semet <notifications@github.com> a
écrit :

**Comment by jaredgrubb** [comment]
Being able to Rebuild+Edit Properties would be amazing! (As would the ability to have a URL pre-fill a form).

File: master/buildbot/data/builds.py [hunk 1]
```diff
@@ -100,12 +102,31 @@ def startConsuming(self, callback, options, kwargs):
                                                  ('builds', str(buildid), None))
 
     def control(self, action, args, kwargs):
-        if action != "stop":
+        # we convert the action into a mixedCase method name
+        action_method = getattr(self, "action" + action[0].upper() + action[1:])
```
Comment (jaredgrubb) [inline master/buildbot/data/builds.py]: FWIW, your code is fine, but you can do "action[0].capitalize()" for the same effect.

**Comment by tardyp** [comment]
addressed review comments, and python 2.6 issues

**Comment by ssavrim** [comment]
:+1:

**Pull request merged**